src/renderers/dom/server/__tests__/ReactServerRendering-test.js

```diff
@@ -400,4 +400,42 @@ describe('ReactServerRendering', function() {
       expect(markup.indexOf('hello, world') >= 0).toBe(true);
     });
   });
+
+  it('warns with a no-op when an async setState or replaceState is triggered', function() {
+    var Component = React.createClass({
+      componentWillMount: function() {
+        this.setState({});
+        setTimeout(() => {
+          this.setState({});
+          setTimeout(() => {
+            this.replaceState({});
+          });
+        });
+      },
+      render: function() {
+        return <div onClick={() => {}} />;
+      },
+    });
+
+    spyOn(console, 'error');
+    ReactServerRendering.renderToString(<Component />);
+    jest.runOnlyPendingTimers();
+    expect(console.error.calls.count()).toBe(1);
+    expect(console.error.calls.mostRecent().args[0])
+    .toBe(
+      'Warning: setState(...): Can only update a mounted or mounting' +
+      ' component. This usually means you called setState() on an' +
+      ' unmounted component. This is a no-op. Please check the code for' +
+      ' the Component component.'
+    );
+    jest.runOnlyPendingTimers();
+    expect(console.error.calls.count()).toBe(2);
+    expect(console.error.calls.mostRecent().args[0])
```

Comment:
The first `setState` is not supposed to trigger warnings. But as you may already have found out, the test just before uses `setState` that way and would fail anyway. So you're right, that's pointless, I just realised !

So I will do exactly as you proposed, and detach multiple tests, that makes total sense! I kinda tried nested timeouts but wasn't really sure it was fine, code-readability-wise !